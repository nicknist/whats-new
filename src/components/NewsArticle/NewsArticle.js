import React from 'react';
import './NewsArticle.css';

// NEWSARTICLE COMPONENT CODE GOES HERE

const NewsArticle = ({ articleInfo }) => {
  return (
    <article>
      <h1>{articleInfo.headline}</h1>
      <img className="headline" src={articleInfo.img} alt="Photo based on Headline" />
      <p>{articleInfo.description}</p>
      <a href={articleInfo.url}> Link to Article</a>
    </article>
  )
}

export default NewsArticle;
