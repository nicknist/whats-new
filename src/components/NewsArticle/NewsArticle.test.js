import React from 'react';
import NewsArticle from './NewsArticle';
import { shallow } from 'enzyme';

describe('NewsArticle', () => {
  it('should create the UI for a news article', () => {
    const articleObject = {
      id: 1,
      headline: 'Mock Headline 1',
      img: 'fakeurl.com',
      description: 'Fake fake stuff.',
      url: 'anotherfakeurl.net'
    }
    const wrapper = shallow(<NewsArticle articleInfo={articleObject}/>)

    expect(wrapper).toMatchSnapshot();
  })
})
