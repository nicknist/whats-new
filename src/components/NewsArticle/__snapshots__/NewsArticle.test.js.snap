// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should create the UI for a news article 1`] = `
<article>
  <h1>
    Mock Headline 1
  </h1>
  <img
    alt=""
    className="headline"
    src="fakeurl.com"
  />
  <p>
    Fake fake stuff.
  </p>
  <a
    href="anotherfakeurl.net"
  >
     Link to Article
  </a>
</article>
`;
